jax/lax/lax.py

```diff
@@ -2433,6 +2433,34 @@ def _conv_general_dilated_batch_rule(
       out = _reshape_axis_into(out_spec[1], out_spec[1] + 1, out)
       return out, out_spec[1]
 
+def _masked(x, logical_shape, dimensions, value=0):
+  if len(dimensions) == 0:
+    return x
+
+  masks = [broadcasted_iota(onp.int32, x.shape, d) < logical_shape[d]
+           for d in dimensions]
+  mask_intersection = masks[0]
+  for mask in masks[1:]:
+    mask_intersection &= mask
+  return select(mask_intersection, x, full_like(x, value))
+
+def _conv_general_dilated_masking_rule(
+        padded_vals, logical_shapes, window_strides, padding, lhs_dilation,
+        rhs_dilation, dimension_numbers, feature_group_count, batch_group_count,
+        lhs_shape, rhs_shape, precision):
+  lhs, rhs = padded_vals
+  logical_lhs_shape, rhs_shape = logical_shapes
```

Comment:
AFAICT `rhs_shape` is ignored. Can you add a check that it isn't polymorphic?

Reply:
Added a check that `rhs_shape` is unpadded, i. e. `logical_shape==padded_shape`. `rhs_shape` can be polymorphic: In the tests it is `(2, 3, i, o)`.